be/test/vec/core/get_common_type_test.cpp

```diff
@@ -143,10 +146,8 @@ INSTANTIATE_TEST_SUITE_P(data_type, LeastSuperTypeTest,
                                  {"Float32 Float64", "Float64"},
                                  {"Float32 UInt16 Int32", "Float64"},
                                  {"Float32 Int16 UInt32", "Float64"},
-                                 {"Date Date", "Date"},
```

Comment:
it's should be fixed at be/src/vec/data_types/get_least_supertype.cpp line:288
add datev2 and datetimev2 
`        if (which.is_date_v2()) {
            *type = std::make_shared<DataTypeDateV2>();
            return;
        }
        if (which.is_date_time_v2()) {
            *type = std::make_shared<DataTypeDateTimeV2>();
            return;
        }`

Reply:
@zhangstar333 related code removed by PR #36743

Reply:
<img width="310" alt="image" src="https://github.com/user-attachments/assets/3a3ab35c-8821-4d36-9a85-ba1cad279f4b">

Reply:
done